ts/src/bitmex.ts

```diff
@@ -428,14 +424,37 @@ export default class bitmex extends Exchange {
                 expiryDatetime = this.safeString (market, 'expiry');
                 expiry = this.parse8601 (expiryDatetime);
                 symbol = symbol + '-' + this.yymmdd (expiry);
+            } else if (index) {
+                // for index markets, default to id
+                symbol = id;
+            }
+            const isInverse = this.safeValue (market, 'isInverse');  // this is true when BASE and SETTLE are same, i.e. BTC/XXX:BTC
+            const isQuanto = this.safeValue (market, 'isQuanto'); // this is true when BASE and SETTLE are different, i.e. AXS/XXX:BTC
+            const inverse = isQuanto || isInverse;
+            const linear = contract ? !inverse : undefined;
+            const basePrecisionString = this.safeString (this.options['currencyPrecisions'], base, '8'); // 8 is temporary, it should be replaced by precision from this.tempCurrencies which should be defined in fC
```

Comment:
in last line of `fetchCurrencies`, there should be:
`this.options['currencyPrecisions'] = currencies;` and then we can access that data inside `fetchMarkets`. otherwise, `this.currencies` is not yet available by the time `fetchMarkets` runs (this can be fixed, but not in scope of this PR imo).